Source/SwiftLintBuiltInRules/Rules/Idiomatic/FileNameRule.swift

```diff
@@ -56,33 +56,98 @@ struct FileNameRule: OptInRule, SourceKitFreeRule {
 }
 
 private class TypeNameCollectingVisitor: SyntaxVisitor {
+    // All of a visited node's ancestor type names if that node is nested, starting with the furthest
+    // ancestor and ending with the direct parent
+    private var ancestorNames: [String] = []
+
+    // All of the type names found in the file
     private(set) var names: Set<String> = []
 
+    // If true, nested types are only allowed in the file name when used by their fully-qualified name
+    // (e.g. `My.Nested.Type` and not just `Type`)
+    private let requireFullyQualifiedNames: Bool
+
+    init(requireFullyQualifiedNames: Bool) {
+        self.requireFullyQualifiedNames = requireFullyQualifiedNames
+        super.init(viewMode: .sourceAccurate)
+    }
+
+    private func addVisitedNodeName(_ name: String) {
```

Comment:
I propose the following signature:

```suggestion
    private func addNames(for decl: some NamedDeclSyntax) {
```

So you can get rid of the `node.name.text` calls.

Reply:
Good suggestion. The only problem is with `func visitPost(_ node: ExtensionDeclSyntax)` since `ExtensionDeclSyntax` doesn't conform to `NamedDeclSyntax` (we grab the "name" via `node.extendedType.trimmedDescription`). I can still create a function that all the nodes that do conform to `NamedDeclSyntax` can call though

Reply:
Actually, I can use a protocol and conform some types to it to make this nicer